azure-proto-compute/Extensions/ResourceGroupExtensions.cs

```diff
@@ -14,8 +15,11 @@ public static class ResourceGroupExtensions
         /// <param name="resourceGroup"> The <see cref="ResourceGroupOperations" /> instance the method will execute against. </param>
         /// <param name="vmName"> The name of the VirtualMachine. </param>
         /// <returns> Returns an object representing the operations that can be performed over a specific <see cref="VirtualMachine" />.</returns>
+        /// <exception cref="ArgumentNullException"> vmName cannot be null. </exception>
         public static VirtualMachineOperations GetVirtualMachineOperations(this ResourceGroupOperations resourceGroup, string vmName)
         {
+            if (string.IsNullOrWhiteSpace(vmName))
+                throw new ArgumentNullException("vmName cannot be null.");
```

Comment:
this would be an ArgumentException since you don't know if its null or whitespace

same comment on nameof(vmName) as above